### PR TITLE
[PeerTube] Support searching for playlists and channels

### PR DESCRIPTION
**Pull request by TobiGr**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).
- [x] I agree to create a pull request for [NewPipe](https://github.com/TeamNewPipe/NewPipe) as soon as possible to make it compatible with the changed API.

### CODE REVIEW

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/peertube/PeertubeParsingHelper.java [hunk 1]
```diff
@@ -101,12 +105,20 @@ public static void collectStreamsFrom(final InfoItemsCollector collector,
...
                 } else {
-                    extractor = new PeertubeStreamInfoItemExtractor(item, baseUrl);
+                    if (isPlaylistInfoItem) {
+                        extractor = new PeertubePlaylistInfoItemExtractor(item, baseUrl);
+                    } else if (isChannelInfoItem) {
+                        extractor = new PeertubeChannelInfoItemExtractor(item, baseUrl);
+                    } else {
+                        extractor = new PeertubeStreamInfoItemExtractor(item, baseUrl);
+                    }
                 }
```
Comment (Stypox) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/peertube/PeertubeParsingHelper.java:122]: No need to nest `if`s again, just do `else if (isPlaylistInfoItem)` directly.

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/peertube/extractors/PeertubeChannelInfoItemExtractor.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import com.grack.nanojson.JsonArray;
+import com.grack.nanojson.JsonObject;
+
+import org.schabi.newpipe.extractor.channel.ChannelExtractor;
+import org.schabi.newpipe.extractor.channel.ChannelInfoItemExtractor;
+import org.schabi.newpipe.extractor.exceptions.ParsingException;
+
+import javax.annotation.Nonnull;
+
+public class PeertubeChannelInfoItemExtractor implements ChannelInfoItemExtractor {
```
Comment (Stypox) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/peertube/extractors/PeertubeChannelInfoItemExtractor.java:12]: ~~Isn't it possible that PeerTube channels have parent channels?~~ Yes but they are not shown in search results even by Peertube websites, so ignore this comment.

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/peertube/extractors/PeertubeChannelInfoItemExtractor.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        }
+        int highestRes = -1;
+        JsonObject avatar = null;
+        for (final Object a: avatars) {
+            if (((JsonObject) a).getInt("width") > highestRes) {
+                avatar = (JsonObject) a;
+                highestRes = avatar.getInt("width");
+            }
+        }
+        return baseUrl + avatar.getString("path");
```
Comment (Stypox) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/peertube/extractors/PeertubeChannelInfoItemExtractor.java:48]:
```suggestion
        return item.getArray("avatars").stream()
                .filter(object -> object instanceof JsonObject)
                .map(object -> (JsonObject) object)
                .max(Comparator.comparingInt(avatar -> avatar.getInt("width")))
                .map(avatar -> baseUrl + avatar.getString("path"))
                .orElse(null);
```

**Review by Stypox — CHANGES_REQUESTED** [review]
Looks almost good to me, thanks :-)

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/peertube/extractors/PeertubeChannelInfoItemExtractor.java (hunk 3 above)
Comment (TobiGr) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/peertube/extractors/PeertubeChannelInfoItemExtractor.java]: That looks way easier. For some reason, I don't think of using streams...

**Review by Stypox — APPROVED** [review]
Thank you! The CI fail is unrelated

**Pull request merged**